homeassistant/components/tts/__init__.py

```diff
@@ -387,25 +391,25 @@ def state(self) -> str | None:
             return None
         return self.__last_tts_loaded
 
-    @property
-    @abstractmethod
```

Comment:
Removing the `abstractmethod` decorator removes the check to implement the property when defining a subclass. We still need to enforce this implementation somehow as it's not optional.

Reply:
But when the subclass sets the `_attr_supported_languages`, then there is no need to overwrite the property.

Reply:
Right. I'm saying we need another way now to enforce the subclass implementing this feature.

Reply:
In the current state it would raise an `AttributeError` when neither the variable is set nor the property is overwritten (as the variable is only there with a type annotation, not a default value). It is for example done like this in the SelectEntity.

https://github.com/home-assistant/core/blob/e4280b2c0013cf0098bbf493f57c9052247442c7/homeassistant/components/select/__init__.py#L165-L168

Reply:
I think the difference there is that a select entity can't be added without a state update that would raise an exception if current option isn't implemented. A tts entity can be added without implementing these features since they aren't accessed as part of the state update.

Reply:
Would the `__init_subclass__` approach be acceptable? With it, a AttributeError will be raised on creating a subclass when the variable and the property are missing. (I will be adding tests of course)

Reply:
Erik and I discussed this and suggest we check if the attribute is implemented in `async_internal_added_to_hass`. Ie the implement that method and do the check there.